main.tf

```
@@ -1 +1,445 @@
+      region         = data.aws_region.current.name
+      ssm_path       = local.ssm_root_path
+      version        = var.vault_version
+    }), "'"])
+  }
+}
+
+# Manage S3 bucket module
+module "s3_bucket" {
+  source  = "terraform-aws-modules/s3-bucket/aws"
+  version = "0.0.1"
+
+  bucket = local.s3_bucket_name
+}
+
+
+resource "aws_s3_bucket_policy" "this" {
+  bucket = module.s3_bucket.id
+  policy = templatefile("${path.module}/policies/bucket_policy.json", { bucket_arn = module.s3_bucket.arn })
+}
+
+# Manage IAM module
+module "iam" {
+  source = "./modules/iam"
+
+  role_name = local.role_name
+  policy_vars = {
+    bucket_name    = module.s3_bucket.id
+    dynamodb_table = local.dynamodb_table
+    kms_key_id     = local.kms_key_id
+    stack_name     = var.name
+    ssm_path       = local.ssm_root_path
+  }
+}
+
+# Generate a random id for each deployment
+resource "random_string" "this" {
+  length  = 8
+  special = false
+  upper   = false
+}
+
+# Manage archive and appscript files
+resource "aws_s3_bucket_object" "salt_zip" {
+  bucket = module.s3_bucket.id
+  key    = local.archive_file_name
+  source = join("/", [local.archive_dir_path, local.archive_file_name])
+  etag   = data.archive_file.salt.output_md5
+}
+
+resource "aws_s3_bucket_object" "configs_zip" {
+  count  = var.vault_configs_path == null ? 0 : 1
+  bucket = module.s3_bucket.id
+  key    = local.configs_file_name
+  source = join("/", [local.archive_dir_path, local.configs_file_name])
+  etag   = data.archive_file.configs[count.index].output_md5
+}
+
+resource "aws_s3_bucket_object" "app_script" {
+  bucket  = module.s3_bucket.id
+  key     = local.appscript_file_name
+  content = data.template_file.appscript.rendered
+  etag    = md5(data.template_file.appscript.rendered)
+}
+
+# Manage KMS key
+resource "aws_kms_alias" "this" {
+  count         = var.kms_key_id == null ? 1 : 0
+  name          = "alias/${var.name}"
+  target_key_id = join("", aws_kms_key.this.*.key_id)
+}
+
+resource "aws_kms_key" "this" {
+  count       = var.kms_key_id == null ? 1 : 0
+  description = "KMS Key for ${var.name}"
+
+  tags = merge({ Name = var.name }, local.tags)
+}
+
+# Manage domain record
+resource "aws_route53_record" "this" {
+  zone_id = var.route53_zone_id
+  name    = local.vault_url
+  type    = "A"
+
+  alias {
+    name                   = aws_lb.this.dns_name
+    zone_id                = aws_lb.this.zone_id
+    evaluate_target_health = false
+  }
+}
+
+# Manage certificate
+resource "aws_acm_certificate" "this" {
+  count = var.certificate_arn == null ? 1 : 0
+
+  domain_name       = local.vault_url
+  validation_method = "DNS"
+
+  tags = merge({ Name = var.name }, local.tags)
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "cert_validation" {
+  count = var.certificate_arn == null ? 1 : 0
+
+  name    = join("", aws_acm_certificate.this.*.domain_validation_options.0.resource_record_name)
```

this looks like tf 0.11 syntax. 
```suggestion
  name    = aws_acm_certificate.this[0].domain_validation_options[0].resource_record_name
```